test/unit/config_parse/lexer/test_config_lexer.cpp

```diff
@@ -0,0 +1,175 @@
+#include "color.hpp"
+#include "lexer.hpp"
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+
+namespace { /*テスト汎用*/
+
+#define ARRAY_SIZE(array) (sizeof(array) / sizeof(array[0]))
+
+struct Result {
+	Result() : is_success(true) {}
+	bool        is_success;
+	std::string error_log;
+};
+
+int GetTestCaseNum() {
+	static unsigned int test_case_num = 0;
+	++test_case_num;
+	return test_case_num;
+}
+
+void PrintOk() {
+	std::cout << utils::color::GREEN << GetTestCaseNum() << ".[OK] " << utils::color::RESET
+			  << std::endl;
+}
+
+void PrintNg() {
+	std::cerr << utils::color::RED << GetTestCaseNum() << ".[NG] " << utils::color::RESET
+			  << std::endl;
+}
+
+void PrintError(const std::string &message) {
+	std::cerr << utils::color::RED << message << utils::color::RESET << std::endl;
+}
+
+} // namespace
+
+/*------------------------------------------------------------------*/
+/*------------------------------------------------------------------*/
+
+/* namespace内で定義しないと他のコードで使えない */
+namespace config {
+namespace node {
+
+bool operator==(const node::Node &lhs, const node::Node &rhs) {
+	return lhs.token == rhs.token && lhs.token_type == rhs.token_type;
+}
+
+bool operator!=(const node::Node &lhs, const node::Node &rhs) {
+	return lhs.token != rhs.token || lhs.token_type != rhs.token_type;
```

Comment:
どっちでも良いのですが、これだけでも良いのかもです
```suggestion
	return !(lhs == rhs);
```

Reply:
この部分は==の実装に依存しない方がいいかなと思ってそうしました

Reply:
了解です！